content/BPIP-4.md

```diff
@@ -0,0 +1,220 @@
+
+## Rationale
+The aforementioned price discovery techniques are widely accepted within the industry as current standards. Our interface has the capability to incorporate any external protocols that utilize these established techniques. As an example, we have described a minimal integration between Boson Protocol and Seaport.
+
+1. Seller creates a BP Offer. Let's say the offer has quantity = 1000, token is USDC, and the price here is set to 0 (0 being a magic number, as the protocol must know this is not a free offer) because the price discovery will happen via Seaport.
+
+  ```javascript
+  offer {
+    ...
+    exchangeToken: USDC address
+    quantityAvailable: 1000
+    type: OfferType.PriceDiscovery
+    price: 0
+  }
+  offerDates: {
+    ...
+    validFrom: 02/06/2023
+    validUntil: 02/06/2024
+  }
+  ```
+2. Seller preMints 1000 vouchers. 
+3. Seller approves protocol to transfer USDC
+4. Seller approves Seaport to transfer preMinted BVs
+5. Seller creates an Order on Seaport with the following fields:
+
+  ```javascript
+    {
+      offer:
+        {
+          itemType: 2 // ERC721
+          token: BV address
+          identifierOrCriteria: the voucher id // check if Seaport has a batch function
+          startAmount: 1
+          endAmount:1
+        }
+      consideration: {
+        itemType: 1 // ERC20
+        token: USDC address
+        identifierOrCriteria: 0
+        startAmount: 10
+        endAmount: 100 // The initial price will be 10 and 100 is the price at the moment the 
+                       // offer expires, realised amount is calculated linearly based on the time 
+                       // elapsed since the order became active.
+      }
+      startTime: protocol offerDates.validFrom
+      endtime: protocol offerDates.validUntil
+      ...
+    }
+  ```
+6. Buyer approves protocol to transfer USDC
+7. Buyer calls `commitToOffer` which the following fields:
+
+```javascript
+const basicOrderParameters = {
+ considerationToken: USDC,
+ considerationAmount: startAmount * t,
+ offerer: seller address,
+ offerToken: BV,
+ offerAmount: 1, 
+ ...
+}
+
+const priceDiscovery = {
+  price: startAmount * t
+  priceDiscoveryContract: seaport address
+  priceDiscoveryData: encodeFunctionData("fulfillBasicOrder", basicOrderParameters)
+}
+
+commitToOffer(..., priceDiscovery)
+```
+
+### Notes
+Note that 2-side AMM pools are out of the scope of this BPIP as they would either require us to make changes to the Boson Voucher, or they would require us to develop an intermediate token. Although this is interesting, it is out of the scope of this proposal. 
+
+## Backward compatibility
+
+This proposed improvement to the protocol is fully backward compatible, we would still support fixed price offers and maintain the existing interfaces
+
+All of the methods described here would be compatible with the current Boson Vouchers. 
+
+## Implementation
+
```

Comment:
Perhaps a note here, that the reviewers have proposed that we need to look at building a few POCs to show that this interface can be used to leverage pre-exisiting price discovery systems. We think that by looking at Seaport, a16z's auctions, sudoswap's AMMs, and building POCs for them, that is the right next step.

Reply:
note added